pwndbg/dbg/__init__.py

```diff
@@ -12,6 +13,169 @@
 dbg: Debugger = None
 
 
+class TypeCode(Enum):
+    """
+    Broad categories of types.
+    """
+
+    POINTER = 1
+    ARRAY = 2
+    STRUCT = 3
+    TYPEDEF = 4
+    UNION = 5
+    INT = 6
+    ENUM = 7
+
+
+class TypeField:
```

Comment:
For my own understanding, this will eventually be something that GDB and LLDB will subclass right? We just don't need to do it at the moment for things to work?

Reply:
Interestingly, it doesn't look like it's going to be necessary to subclass it later on

From my work on the later changes, LLDB can fill most of them exactly, and two of them (the ones relating to enum fields and inherited members) I haven't implemented yet, but apparently can be implemented with very little effort